src/IonTimestamp.ts

```diff
@@ -310,154 +281,173 @@ export class Timestamp {
+                if(!this.fraction.equals(expected.fraction)) return false;
+            case Precision.SECONDS:
+                if(this.seconds !== expected.seconds) return false;
+            case Precision.HOUR_AND_MINUTE:
+                if(this.minute !== expected.minute || this.hour !== expected.hour) return false;
+            case Precision.DAY:
+                if(this.day !== expected.day) return false;
+            case Precision.MONTH:
+                if(this.month !== expected.month) return false;
+            case Precision.YEAR:
+                if(this.year !== expected.year) return false;
         }
-      }
+        return true;
     }
-  }
-
-  equals(expected : Timestamp) : boolean {//TODO implement instant equals https://github.com/amzn/ion-js/issues/132
-    return this.getPrecision() === expected.getPrecision() && this.offset === expected.offset && this.dataModelEquals(expected);
-  }
-
-  dataModelEquals(expected : Timestamp) : boolean {
-      switch (this.precision) {
-          case Precision.NULL:
-              return expected.precision === Precision.NULL;
-          case Precision.SECONDS:
-              if(!this.seconds.equals(expected.seconds)) return false;
-          case Precision.HOUR_AND_MINUTE:
-              if(this.minute !== expected.minute || this.hour !== expected.hour) return false;
-          case Precision.DAY:
-              if(this.day !== expected.day) return false;
-          case Precision.MONTH:
-              if(this.month !== expected.month) return false;
-          case Precision.YEAR:
-              if(this.year !== expected.year) return false;
-      }
-      return true;
-  }
-
-  stringValue() : string {
-    let image: string;
-    let t: Timestamp = this;
-
-    switch (t.precision) {
-      default: throw { msg: "invalid value for timestamp precision", where: "IonValueSupport.timestamp.toString" };
-      case Precision.NULL:
-        return "null.timestamp";
-      case Precision.SECONDS:
-        //formats decimal to timestamp second, adds a leading 0 and/or cuts off the trailing period
-        image = t.seconds.toString();
-        if(image.charAt(1)  === '.') image = "0" + image;
-        if(image.charAt(image.length - 1) === '.') image = image.slice(0, image.length - 1);
-      case Precision.HOUR_AND_MINUTE:
-        image = _to_2_digits(t.minute) + (image ? ":" + image : "");
-        image = _to_2_digits(t.hour) + (image ? ":" + image : "");
-      case Precision.DAY:
-        image = _to_2_digits(t.day) + (image ? "T" + image : "T");
-      case Precision.MONTH:
-        image = _to_2_digits(t.month) + (image ? "-" + image : "");
-      case Precision.YEAR:
-        if (t.precision === Precision.YEAR) {
-          image = to_4_digits(t.year) + "T";
-        } else if (t.precision === Precision.MONTH) {
-          image = to_4_digits(t.year) + "-" + image + "T";
-        } else {
-          image = to_4_digits(t.year) + "-" + image;
+
+    stringValue() : string {
```

Comment:
What's the reason you always "prepend" i.e. `image = <new part> + image`?  That just seems weird to me. 

I would expect that the cases appear in the reverse and and new parts are appended--i..e.:  `image = image + <new part>`.  I suspect that this would be simpler as well...